tests/prague/eip7692_eof_v1/eip7620_eof_create/test_subcontainer_validation.py

```diff
@@ -0,0 +1,247 @@
+    [eofcreate_code_section, returncontract_code_section],
+    ids=["eofcreate", "returncontract"],
+)
+def test_reverting_container(
+    eof_state_test: EOFStateTestFiller,
+    zero_section: Container,
+):
+    """Test revert containers"""
+    eof_state_test(
+        deploy_tx=zero_section == returncontract_code_section,
+        data=Container(
+            sections=[
+                zero_section,
+                revert_sub_container,
+            ],
+        ),
+        container_post=Account(storage={slot_code_worked: value_code_worked}),
+    )
+
+
+@pytest.mark.parametrize(
+    "code_section,first_sub_container",
+    [
+        (eofcreate_code_section, returncontract_sub_container),
+        (returncontract_code_section, stop_sub_container),
+    ],
+    ids=["eofcreate", "returncontract"],
+)
+@pytest.mark.parametrize(
+    "extra_sub_container",
+    [stop_sub_container, revert_sub_container, returncontract_sub_container],
+    ids=["stop", "revert", "returncontract"],
+)
+def test_orphan_container(
+    eof_test: EOFTestFiller,
+    code_section: Section,
+    first_sub_container: Container,
+    extra_sub_container: Container,
+):
+    """Test orphaned containers"""
+    eof_test(
+        deploy_tx=code_section == returncontract_code_section,
+        data=Container(
+            sections=[
+                code_section,
+                first_sub_container,
+                extra_sub_container,
+            ],
+        ),
+        kind="initcode",
+        expect_exception=EOFException.ORPHAN_SUBCONTAINER,
+    )
+
+
+@pytest.mark.parametrize(
+    "code_section,sub_container",
+    [
+        pytest.param(
+            eofcreate_code_section,
+            returncontract_sub_container,
+            id="EOFCREATE/RETURNCONTRACT",
+        ),
+        pytest.param(returncontract_code_section, stop_sub_container, id="RETURNCONTRACT/STOP"),
+        pytest.param(
+            returncontract_code_section, return_sub_container, id="RETURNCONTRACT/RETURN"
+        ),
+        pytest.param(eofcreate_code_section, revert_sub_container, id="EOFCREATE/REVERT"),
+        pytest.param(
+            returncontract_code_section, revert_sub_container, id="RETURNCONTRACT/REVERT"
+        ),
+    ],
+)
+def test_container_combos_valid(
+    eof_state_test: EOFStateTestFiller,
+    code_section: Section,
+    sub_container: Container,
+):
+    """Test valid subcontainer reference / opcode combos"""
+    eof_state_test(
+        deploy_tx=code_section == returncontract_code_section,
+        data=Container(
+            sections=[
+                code_section,
+                sub_container,
+            ],
+        ),
+        container_post=Account(storage={slot_code_worked: value_code_worked}),
+    )
+
+
+@pytest.mark.parametrize(
+    "code_section,first_sub_container,error",
+    [
+        pytest.param(
+            eofcreate_code_section,
+            stop_sub_container,
+            EOFException.UNDEFINED_EXCEPTION,
```

Comment:
These exceptions might appear when this branch is rebased and we send the `--initcode` flag to evmone.